components/album/SubAlbum.vue

```diff
@@ -14,40 +15,70 @@ const emit = defineEmits<{ expand: [] }>();
 function expand() {
   emit("expand");
 }
+const childTracks = computed(() => {
+  if (album.value == null || !album.value.children) return [];
+
+  return album.value.children.filter(
+    (c): c is TrackModel => c.type === "track",
+  );
+});
 </script>
 
 <template>
   <section
-    class="relative my-4 mr-3 cursor-pointer gap-2 duration-150"
-    :class="active ? 'rounded-3xl border-2 p-5 shadow-lg' : ''"
+    class="block relative mr-3 cursor-pointer gap-2 duration-150 my-4"
+    :class="active ? 'rounded-3xl p-6 album-shadow' : 'group'"
     @click.stop="expand"
   >
     <div
-      :class="
-        active
-          ? ''
-          : 'bg-slate-100 absolute -inset-x-3 -inset-y-2 rounded-xl opacity-0'
-      "
+      v-if="!active"
+      class="absolute -inset-4 opacity-0 group-hover:opacity-100 bg-background-2 rounded-2xl"
     ></div>
-    <section
-      v-if="album"
-      class="relative flex items-center justify-between gap-2"
-    >
-      <CoverImage :src="album.cover" class="w-20 rounded-md" />
-      <p class="text-2xl font-bold">{{ album.title }}</p>
-      <p v-if="album?.children" class="ml-auto">
-        {{ t("collection.track-count", album?.children.length) }}
-      </p>
+    <section v-if="album" class="relative flex items-center gap-4">
+      <CoverImage :src="album.cover" class="w-[68px] rounded-md" />
+      <div class="text-2xl font-extrabold">{{ album.title }}</div>
+      <div
+        v-if="album?.children && !active"
+        class="flex gap-1.5 ml-auto px-4 py-2 bg-background-2 rounded-3xl border-[1px] border-background-2 group-hover:border-label-separator"
+      >
+        <span class="font-semibold text-lg leading-6">{{
```

Comment:
Do we currently only allow for albums containing sub-albums or do we have deeper nested structure? I guess we agreed on that we won't. In this case this code is fine. Otherwise we'd have to improve here.

Reply:
I guess this discussion has been confirmed earlier, we have just one level of nesting here.